[#4394] *Really* hide inactive tabs on the main screen (using zIndex).

mandrigin opened the pull request:
fixes #4394 

Otherwise, leftovers from other screens (like images and labels) are still present and preventing from tapping certain areas.

status: ready <!-- Can be ready or wip -->

rasom commented:
apk uploaded to https://i.diawi.com/J5rpfh for branch PR-4397

<blockquote><div><strong><a href="https://i.diawi.com/J5rpfh">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

rasom commented:
ipa uploaded to https://i.diawi.com/LZGx6t for branch PR-4397

<blockquote><div><strong><a href="https://i.diawi.com/LZGx6t">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

lukaszfryc commented:
@mandrigin iOS works fine but Android throws this error after logging in:

```
05-23 15:20:51.197  4154  4433 D NetworkController.MobileSignalController(0/-2): getMobileIconGroup(): 0
05-23 15:20:51.198 24724 24786 I TESTFAIRYSDK: Saved stack trace to /data/user/0/im.status.ethereum/files/testfairy-stacktrace.97521313-4803622-8311330-418654802-11d787301fcc0b520056c9c4b49f57ed3e1f5569.1527081651198
05-23 15:20:51.198 24724 24786 I TESTFAIRYSDK: Sending crash {"timestamp":1527081651,"message":"java.lang.IllegalArgumentException: width and height must be > 0","stackTrace":"java.lang.IllegalArgumentException: width and height must be > 0\n\tat android.graphics.Bitmap.createBitmap(Bitmap.java:969)\n\tat android.graphics.Bitmap.createBitmap(Bitmap.java:948)\n\tat android.graphics.Bitmap.createBitmap(Bitmap.java:915)\n\tat com.horcrux.svg.SvgViewShadowNode.drawOutput(SvgViewShadowNode.java:130)\n\tat com.horcrux.svg.SvgViewShadowNode.onCollectExtraUpdates(SvgViewShadowNode.java:120)\n\tat com.facebook.react.uimanager.ReactShadowNodeImpl.dispatchUpdates(ReactShadowNodeImpl.java:325)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:968)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.applyUpdatesRecursive(UIImplementation.java:959)\n\tat com.facebook.react.uimanager.UIImplementation.updateViewHierarchy(UIImplementation.java:700)\n\tat com.facebook.react.uimanager.UIImplementation.dispatchViewUpdates(UIImplementation.java:654)\n\tat com.facebook.react.uimanager.UIManagerModule.onBatchComplete(UIManagerModule.java:663)\n\tat com.facebook.react.bridge.NativeModuleRegistry.onBatchComplete(NativeModuleRegistry.java:136)\n\tat com.facebook.react.bridge.CatalystInstanceImpl$BridgeCallback.onBatchComplete(CatalystInstanceImpl.java:145)\n\tat com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)\n\tat android.os.Handler.handleCallback(Handler.java:751)\n\tat android.os.Handler.dispatchMessage(Handler.java:95)\n\tat com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)\n\tat android.os.Looper.loop(Looper.java:154)\n\tat com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:194)\n\tat java.lang.Thread.run(Thread.java:762)\n"}
05-23 15:20:51.201 24724 25039 I System.out: Thread-2876(ApacheHTTPLog):isSBSettingEnabled false
05-23 15:20:51.205 24724 25039 I System.out: Thread-2876(ApacheHTTPLog):isShipBuild true
05-23 15:20:51.206 24724 25039 I System.out: Thread-2876(ApacheHTTPLog):getDebugLevel 0x4f4c
05-23 15:20:51.206 24724 25039 I System.out: Thread-2876(ApacheHTTPLog):Smart Bonding Setting is false
05-23 15:20:51.206 24724 25039 I System.out: Thread-2876(ApacheHTTPLog):SmartBonding Setting is false, SHIP_BUILD is true, log to file is false, DBG is false, DEBUG_LEVEL (1-LOW, 2-MID, 3-HIGH) is 1
05-23 15:20:51.208  3370  3814 D EnterpriseController: netId is 0
```
<img src="https://user-images.githubusercontent.com/7532782/40427141-25dc615c-5e9d-11e8-90bf-24c2c740623a.png" width="200">

mandrigin commented:
@lukaszfryc gotcha, I'll go check it

flexsurfer commented:
@mandrigin just to make sure, opacity used to do not re-render component, are you sure changing display value do not lead to component rerender?

mandrigin commented:
@flexsurfer it looks like this approach has too many more potential regressions, I'd better avoid it for now...

mandrigin committed:
[#4394] Use zIndex to move inactive tabs back.

That ensures that no view from an inactive tab will interfere with the
active one.

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

mandrigin commented:
@flexsurfer updated to zIndex change. that looks much safer IMO.

rasom commented:
apk uploaded to https://i.diawi.com/BBkRQc for branch PR-4397

<blockquote><div><strong><a href="https://i.diawi.com/BBkRQc">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

rasom commented:
ipa uploaded to https://i.diawi.com/9XoW9q for branch PR-4397

<blockquote><div><strong><a href="https://i.diawi.com/9XoW9q">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

rasom commented:
apk uploaded to https://i.diawi.com/5bcaUv for branch PR-4397

<blockquote><div><strong><a href="https://i.diawi.com/5bcaUv">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

rasom commented:
ipa uploaded to https://i.diawi.com/vUa48o for branch PR-4397

<blockquote><div><strong><a href="https://i.diawi.com/vUa48o">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

asemiankevich commented:
Looks good to me having more than 10 items on Home , able to tap any from the list without any issues. Can login / logout both devices (iOS bug is fixed)
